File: grpc-trading-platform/user-service/src/main/java/org/hle/user/service/handler/UserInformationRequestHandler.java
package org.hle.user.service.handler;

import org.hle.user.UserInformation;
import org.hle.user.UserInformationRequest;
import org.hle.user.exception.UnknownUserException;
import org.hle.user.repository.PortfolioRepository;
import org.hle.user.repository.UserRepository;
import org.hle.user.util.EntityMessageMapper;
import org.springframework.stereotype.Service;

@Service
public class UserInformationRequestHandler {
    private final UserRepository userRepository;
    private final PortfolioRepository portfolioRepository;

    public UserInformationRequestHandler(UserRepository userRepository, PortfolioRepository portfolioRepository) {
        this.userRepository = userRepository;
        this.portfolioRepository = portfolioRepository;
    }

    public UserInformation getUserInformation(UserInformationRequest request) {
        var user = userRepository.findById(request.getUserId())
                .orElseThrow(() -> new UnknownUserException(request.getUserId()));

        var portfolioItems = portfolioRepository.findAllByUserId(user.getId());

        return EntityMessageMapper.toUserInformation(user, portfolioItems);
    }
}

File: grpc-trading-official/user-service/src/main/kotlin/org/hle/userservice/handler/UserInformationRequestHandler.kt
package org.hle.userservice.handler

import org.hle.user.UserInformation
import org.hle.user.UserInformationRequest
import org.hle.userservice.entity.User
import org.hle.userservice.entity.PortfolioItem
import org.hle.userservice.exception.UnknownUserException
import org.hle.userservice.repository.PortfolioItemRepository
import org.hle.userservice.repository.UserRepository
import org.hle.userservice.util.EntityMessageMapper
import org.springframework.stereotype.Service

@Service
class UserInformationRequestHandler(private val userRepository: UserRepository,
                                    private val portfolioRepository: PortfolioItemRepository
) {

    fun getUserInformation(request: UserInformationRequest): UserInformation {
        val user: User =
            userRepository.findById(request.getUserId())
                .orElseThrow { UnknownUserException(request.getUserId()) }

        val portfolioItems: List<PortfolioItem> =
            portfolioRepository.findByUserId(user.id!!)

        return EntityMessageMapper.toUserInformation(user, portfolioItems.toMutableList())
    }
}
